packages/flutter/lib/src/services/spell_check.dart

```diff
@@ -0,0 +1,344 @@
+        Offset selectionMidpoint, double textLineHeight);
+
+    // Build TextSpans with misspelled words indicated.
+    TextSpan buildTextSpanWithSpellCheckSuggestions(
+        List<SpellCheckerSuggestionSpan>? spellCheckResults,
+        TextEditingValue value, TextStyle? style, bool composingWithinCurrentTextRange);
+}
+
+class DefaultSpellCheckSuggestionsHandler implements SpellCheckSuggestionsHandler {
+    int scssSpans_consumed_index = 0;
+    int text_consumed_index = 0;
+    TextSpan ts = TextSpan();
+
+    final TargetPlatform platform;
+
+    DefaultSpellCheckSuggestionsHandler(this.platform);
+
+    @override
+    Widget buildSpellCheckSuggestionsToolbar(
+        List<SpellCheckerSuggestionSpan>? spellCheckResults,
+        TextSelectionDelegate delegate, 
+        List<TextSelectionPoint> endpoints, Rect globalEditableRegion, 
+        Offset selectionMidpoint, double textLineHeight) {
+            return _SpellCheckerSuggestionsToolbar(
+            platform: platform,
+            delegate: delegate,
+            endpoints: endpoints,
+            globalEditableRegion: globalEditableRegion,
+            selectionMidpoint: selectionMidpoint,
+            textLineHeight: textLineHeight,
+            spellCheckerSuggestionSpans: spellCheckResults,
+            );
+        }
+
+  @override
+  TextSpan buildTextSpanWithSpellCheckSuggestions(
+      List<SpellCheckerSuggestionSpan>? spellCheckResults,
+      TextEditingValue value, TextStyle? style, bool composingWithinCurrentTextRange) {
+      scssSpans_consumed_index = 0;
+      text_consumed_index = 0;
+
+      TextStyle misspelledStyle;
+
+      switch(platform) {
+          case TargetPlatform.android:
+          default:
+            misspelledStyle = TextStyle(decoration: TextDecoration.underline,
+                            decorationColor: Colors.red,
+                            decorationStyle: TextDecorationStyle.wavy);
+            break;
+      }
+
+      if (composingWithinCurrentTextRange) {
+          return TextSpan(
+              style: style,
+              children: buildSubtreesWithMisspelledWordsIndicated(spellCheckResults ?? <SpellCheckerSuggestionSpan>[], value.text, style, misspelledStyle, false)
+          );
+      } else {
+          return TextSpan(
+              style: style,
+              children: <TextSpan>[
+                  TextSpan(children: buildSubtreesWithMisspelledWordsIndicated(spellCheckResults ?? <SpellCheckerSuggestionSpan>[], value.composing.textBefore(value.text), style, misspelledStyle, false)),
+                  TextSpan(children: buildSubtreesWithMisspelledWordsIndicated(spellCheckResults ?? <SpellCheckerSuggestionSpan>[], value.composing.textInside(value.text), style, misspelledStyle, true)),
+                  TextSpan(children: buildSubtreesWithMisspelledWordsIndicated(spellCheckResults ?? <SpellCheckerSuggestionSpan>[], value.composing.textAfter(value.text), style, misspelledStyle, false)),
```

Comment:
Did you split this into 3 for debugging or do you plan to keep it like this?  There might be cleaner ways to do this all but maybe I should hold off on reviewing that until it's all working.

Reply:
I planned to keep it this way, but I agree it's not the cleanest way, and I also just came up with a solution for the asynchrony issues I've been having that concerns removing this split. I'll work on that and report back!